Python/Product/TestAdapter/TestContainerDiscovererWorkspace.cs

```diff
@@ -173,7 +176,9 @@ class TestContainerDiscovererWorkspace : ITestContainerDiscoverer, IDisposable {
                 _testFilesUpdateWatcher.AddDirectoryWatch(workspace.Location);
                 oldWatcher?.Dispose();
 
-                var files = Directory.EnumerateFiles(workspace.Location, "*.*", SearchOption.AllDirectories);
+                var files = Directory.EnumerateFiles(workspace.Location, "*.py", SearchOption.AllDirectories)
```

Comment:
I think you should save the results of `interpreterRegistryService.Configurations.Where(x => PathUtils.IsSubpathOf(projectInfo.ProjectHome, x.InterpreterPath))` here in a variable and reuse it, so you don't have to recalculate that for potentially 100s or 1000s of files.

Reply:
Updated